src/DocumentationViewer.js

```diff
@@ -0,0 +1,219 @@
+import Gtk from "gi://Gtk";
+import Gio from "gi://Gio"
+import GLib from "gi://GLib";
+import Adw from "gi://Adw";
+import WebKit from "gi://WebKit"
+
+import resource from "./DocumentationViewer.blp";
+
+Gio._promisify(
+  WebKit.WebView.prototype,
+  "evaluate_javascript",
+  "evaluate_javascript_finish"
+);
+
+Gio._promisify(
+  Gio.File.prototype,
+  "enumerate_children_async",
+  "enumerate_children_finish",
+);
+
+Gio._promisify(
+  Gio.FileEnumerator.prototype,
+  "next_files_async",
+  "next_files_finish",
+);
+
+Gio._promisify(
+  Gio.File.prototype,
+  "load_contents_async",
+  "load_contents_finish",
+);
+
+export default function DocumentationViewer({
+  window:application_window,
+  application
+  }) {
+  const builder = Gtk.Builder.new_from_resource(resource);
+  const window = builder.get_object("documentation_viewer");
+
+  const container = builder.get_object("webview_container");
+  const webview = new WebKit.WebView();
+  const listbox = builder.get_object("listbox");
+
+  const sidebar_button = builder.get_object("sidebar_button");
+  const search_button = builder.get_object("search_button");
+  const search_entry = builder.get_object("search_entry");
+  const search_bar = builder.get_object("search_bar");
+  const back = builder.get_object("back");
+  const forward = builder.get_object("forward");
+
+  const base_path = "/app/share/doc";
+
+  webview.load_uri(`file://${base_path}/gtk4/index.html`);
+  container.child = webview;
+  let loaded = false;
+
+  webview.connect("load-changed", (view, load_event) => {
+    if (load_event === WebKit.LoadEvent.FINISHED) {
+      loaded = true;
+      sidebar_button.active = false;
+    } else {
+      loaded = false;
+    }
+  });
+
+  back.connect("clicked", () => {
+    webview.go_back();
+  });
+
+  forward.connect("clicked", () => {
+    webview.go_forward();
+  });
+
+  sidebar_button.connect("toggled", () => {
+    if (loaded) {
+      if (sidebar_button.active) disableDocSidebar(webview);
+      else enableDocSidebar(webview);
+    }
+  });
+
+  sidebar_button.connect("toggled", () => {
+    if (loaded) {
+      if (sidebar_button.active) disableDocSidebar(webview);
+      else enableDocSidebar(webview);
+    }
+  });
+
+  search_entry.connect("search-changed", () => {
+    listbox.invalidate_filter();
+  });
+
+  search_bar.bind_property(
+    "search-mode-enabled",
+    search_button,
+    "active",
+    GObject.BindingFlags.BIDIRECTIONAL,
+  );
+
+  search_button.connect("toggled", () => {
+    if (search_button.active) sidebar_button.active = true;
+  });
+
+  listbox.set_sort_func(sort);
+  listbox.invalidate_sort();
+  listbox.set_filter_func(filter);
+
+  getDocs(base_path)
+    .then((docs) => {
+      for (const doc in docs) {
+        const row = new Adw.ActionRow({
+          title: doc,
+        });
+        row.add_suffix(new Gtk.Image({ icon_name: "go-next-symbolic" }));
+        listbox.append(row);
+      }
+      listbox.connect("row-selected", (_, row) => {
```

Comment:
This is the convention in Workbench

```suggestion
      listbox.connect("row-selected", (self, row) => {
```